src/circuit.rs

```diff
@@ -244,6 +244,45 @@ impl<'r, F: Field> Region<'r, F> {
     }
 }
 
+/// A lookup table in the circuit.
+///
+/// TODO: Add wrapper type to `ConstraintSystem` to allow us to only deal with lookup
+/// columns here.
```

Comment:
This is a safety concern: without this type separation, chip developers can ignore `Layouter::assign_table` and continue to use `Layouter::assign_region` which does not perform default-value-filling. We don't need to implement this before we merge the Action circuit PR for Orchard, but we _should_ do this before we start encouraging others to use the `halo2` crate.

Reply:
I added a `TableColumn` struct to address this.